dune/grid/common/ZoltanPartition.cpp

```diff
@@ -122,11 +128,62 @@ std::vector<int> zoltanGraphPartitionGridOnRoot(const CpGrid& cpgrid,
         }
 #endif
     }
-    cc.broadcast(&parts[0], parts.size(), root);
+    // free space allocated for zoltan.
     Zoltan_LB_Free_Part(&exportGlobalGids, &exportLocalGids, &exportProcs, &exportToPart);
     Zoltan_LB_Free_Part(&importGlobalGids, &importLocalGids, &importProcs, &importToPart);
     Zoltan_Destroy(&zz);
-    return parts;
+
+    cc.broadcast(&parts[0], parts.size(), root);
+    std::vector<int> my_well_indices;
+
+    if( !pretendEmptyGrid)
+    {
+        std::vector<MPI_Request> reqs(cc.size(), MPI_REQUEST_NULL);
+        my_well_indices = wells_on_proc[root];
+        for ( int i=0; i < cc.size(); ++i )
+        {
+            if(i==root)
+            {
+                continue;
+            }
+            MPI_Isend(wells_on_proc[i].data(), wells_on_proc[i].size(),
+                      MPI_INT, i, 267553, cc, &reqs[i]);
```

Comment:
Then I think it should be named since it is used in two places, also a name would make it clear that it is essentially a random unique number.